### PR TITLE
Adding a note on membership exiry dates

### PR DESCRIPTION
**Pull request by bradleysigma**
(hopefully)

### CODE REVIEW

File: uqcs/app.py [hunk 1]
```diff
@@ -76,6 +77,17 @@ def user_from_request(req):
     else:
         return m.Member(**info), "Success"
 
+def calc_expiries():
+    curr_year = date.today().year
+    expiry_today = f"Feb 29th, {curr_year+1}" if (curr_year%4 == 4-1
+                                                and curr_year%100 != 100-1
+                                                or curr_year%400 == 400-1) else f"Feb 28th, {curr_year+1}"
```
Comment (nicklambourne) [inline uqcs/app.py]: https://docs.python.org/3/library/calendar.html#calendar.isleap

File: uqcs/app.py [hunk 2]
```diff
@@ -76,6 +77,17 @@ def user_from_request(req):
...
+    start_future = f"Jan 1st, {curr_year+1}"
+    expiry_future = f"Feb 29th, {curr_year+2}" if (curr_year%4 == 4-2
```
Comment (nicklambourne) [inline uqcs/app.py]: Horizontal spacing around `%` and all other binary operators should be ` op `.

**Comment by nicklambourne** [comment]
Why is expiry 28th/29th of Feb? We've always rolled over around new year (i.e. expiry 31st Dec) which avoids any leap year nonsense and divorces membership from market day nonsense. It also gives exisiting members a chance to sign up after new year but before market day.

**Review by nicklambourne — CHANGES_REQUESTED** [review]

**Comment by bradleysigma** [comment]
From the Constitution:

>5.2 The membership fees shall be payable upon joining the Society. The payment of the membership subscription means that the member is considered a financial member of the club for the period of twelve (12) months, from the first day of March of the year of subscription.

**Comment by nicklambourne** [comment]
Is this part of the new constitutional changes?

**Comment by bradleysigma** [comment]
Yes.

**Comment by bradleysigma** [comment]
In part, this means that we don't violate the 30 membership requirement on Jan 1st.  
(Also the requirement for committee to be financial members.)

**Comment by nicklambourne** [comment]
This is going to be a goddamn mess.

**Review by nicklambourne — APPROVED** [review]
Nice 👌

**Pull request merged**